### DISCUSSION TITLE
How to visualize the pipelines for time-series models?

### DISCUSSION THREAD

**Discussion by biv3k** [pycaret/pycaret#3200] [Q&A]
Is there a way to visualize the pipeline for time series models? I was experimenting with the "xdg_cds_lt" model and I wanted to see what steps it performs internally?

**Comment by ngupta23** [comment]
You can use `exp.pipeline` to see the generic pipeline structure. Then whatever model you build (e.g. ETS in this case) will get added in place of the DummyForecaster. All this is handled internally in pycaret until you finalize the model. Once you finalize, you get the full pipeline back since the assumption is that until that point, you are still experimenting and you will only put in production after finalizing the model.

![image](https://user-images.githubusercontent.com/33585645/209447044-55741b0b-9dd7-4f6f-b703-deeecc42c506.png)

**Reply by biv3k** [reply]
Thank you!

Do some of the models have tdhe pipeline attribute not defined? For example, I wanted to see how "xgboost_cds_dt" or "lightgbm_cds_dt" models de-seasonalize and de-trend the data. But, it says they are not available.

![image](https://user-images.githubusercontent.com/20134811/209474955-14009d56-f5cf-4bea-b029-e7c8e8453c45.png)

**Reply by biv3k** [reply]
Oh, looks like the finalized model itself is the pipeline. 

![image](https://user-images.githubusercontent.com/20134811/209475387-66962220-51d3-4f60-ae68-bd95fb27489b.png)
